platform/persistence/platform-persistence-core-impl/src/main/java/org/codice/ddf/persistence/internal/PersistentStoreImpl.java

```diff
@@ -289,26 +264,25 @@ public int delete(String type, String cql) throws PersistenceException {
         return idsToDelete.size();
     }
 
-    private SolrClient getSolrCore(String storeName) {
-        if (coreSolrClients.containsKey(storeName)) {
-            LOGGER.debug("Returning core {} from map of coreSolrClients", storeName);
-            return coreSolrClients.get(storeName);
+    private SolrClient getSolrClient(String storeName) {
+        if (solrClients.containsKey(storeName)) {
+            LOGGER.debug("Returning core {} from map of solrClients", storeName);
+            return solrClients.get(storeName);
         }
 
         // Must specify shard in URL so proper core is used
-        SolrClient coreSolrClient = null;
+        SolrClient solrClient = null;
         try {
-            Future<SolrClient> coreSolrClientFuture =
-                    HttpSolrClientFactory.getHttpSolrClient(solrUrl.getResolvedString(), storeName);
-            coreSolrClient = coreSolrClientFuture.get(30, TimeUnit.SECONDS);
-            coreSolrClients.put(storeName, coreSolrClient);
+            Future<SolrClient> clientFuture = clientFactory.newClient(storeName);
+            solrClient = clientFuture.get(30, TimeUnit.SECONDS);
```

Comment:
I think we should change the `solrClients` map to be a map of futures instead of clients so we can remove this blocking 30 second get and allow the retry logic to work.  That would make it work more like the remote Solr catalog provider.

Though I think we should do this eventually, this does not need to be done as part of this PR if you think it is too big of a change since it is out of scope.